openedx/core/djangoapps/content_libraries/management/commands/migrate_lib_to_learning_core.py

```diff
@@ -0,0 +1,234 @@
+                    namespace=COMPONENT_NAMESPACE,
+                    type=block_type,
+                    local_key=block_id,
+                    title=display_name,
+                    created=now,
+                    created_by=None,
+                )
+                published_component_pks[file_path] = component.pk
+                text_content, _created = contents_api.get_or_create_text_content_from_bytes(
+                    learning_package_id=learning_package.id,
+                    data_bytes=xml_bytes,
+                    mime_type=f"application/vnd.openedx.xblock.v1.{block_type}+xml",
+                    created=now,
+                )
+                components_api.add_content_to_component_version(
+                    component_version,
+                    raw_content_id=text_content.pk,
+                    key="definition.xml",
+                    learner_downloadable=False
+                )
+            # Publish all the Draft versions we created.
+            publishing_api.publish_all_drafts(
+                learning_package_id=learning_package.id,
+                message="Initial import from Blockstore",
+                published_at=now,
+            )
+
+            # Now grab the draft version from blockstore, and copy those...
+            draft_files = get_bundle_files_dict(lib.bundle_uuid, use_draft=lib_constants.DRAFT_NAME)
+            draft_definition_files = {
+                file_path: metadata
+                for file_path, metadata in draft_files.items()
+                if file_path.endswith("definition.xml")
+            }
+            for file_path, draft_metadata in draft_definition_files.items():
+                published_metadata = published_metadata_dict.get(file_path)
+                if draft_metadata.modified:
+                    block_type, block_id, _def_xml = file_path.split('/')
+                    xml_bytes = get_bundle_file_data(bundle.uuid, file_path, use_draft=lib_constants.DRAFT_NAME)
+                    display_name = extract_display_name(xml_bytes, file_path)
+
+                    # If this is newly created in the draft, we have to create a
+                    # whole new Component...
+                    if published_metadata is None:
+                        component = components_api.create_component(
+                            learning_package_id=learning_package.id,
+                            namespace=COMPONENT_NAMESPACE,
+                            type=block_type,
+                            local_key=block_id,
+                            created=now,
+                            created_by=None,
+                        )
+                        component_pk = component.pk
+                        version_num = 1
+                    # Otherwise, it's just been modified...
+                    else:
+                        component_pk = published_component_pks[file_path]
+                        version_num = 2
```

Comment:
Agreed.

@ormsbee I don't want to block this PR, so consider adding something like `create_next_component_version` to the API in order to smooth out the I-just-want-to-create-the-next-version use case, while leaving `create_component_version` as-is for advanced cases.

Reply:
I will, thanks. One of the reasons I kept it explicit is because I wanted to have a pattern where the caller has to explicitly say what version they think it should be in order to catch race conditions with multiple writers and force one of them to figure it out rather than to silently overwrite. But it's just clunky when used in this scenario.